### ISSUE TITLE
Articles get 88 score in Lighthouse accessibility audit

### ISSUE THREAD

**Issue by tunetheweb** [GoogleChrome/web.dev#8320]
**Describe the bug**

This appears to be due to two items:
1. Our Hamburger menu button on mobile doesn't have an accessible name - easily fixed with an aria-label.
2. Our Article tags are in a `role="list"` but there are no list items within that list (as it's a `div` instead of `ul`, and there are no `li` items). Probably easiest fix is to just remove the `role="list"` for now as changing the `div` to `ul` causes an indentation, it would require wrapping the links in `<li>`s (possibly requiring further CSS changes), and other "lists" like language links do are not in `role="lists"` anyway.

**To Reproduce**
1. Go to https://web.dev/measure/?url=https%3A%2F%2Fweb.dev%2Fbrowser-level-image-lazy-loading%2F
3. Run Lighthouse Audit
4. Notice we score 88 for a couple of simple markup reasons

**Expected behavior**
As a best practice website, we should be looking to score 100 in a11y audits.

**Screenshots**
N/A

Happy to throw in a PR for this as seems fairly simple fixes.
